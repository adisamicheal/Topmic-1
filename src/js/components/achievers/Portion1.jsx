import React from 'react';
import { shape, arrayOf, string, number } from 'prop-types';
import Part from './Part';

class Portion1 extends React.Component {
	renderArrayHandler() {
		const { achieversData } = this.props;

		return (
			<div>
				<h2>{achieversData.class17Data.textYear}</h2>
				<div className="box" />
				{achieversData.class17Data.data.map(item => (
					<Part
						key={item.id}
						text={item.text}
						textHeading={item.textHeading}
						image={item.image}
					/>
				))}
				<div className="clear" />
			</div>
		);
	}
	render() {
		return <div className="portion1">{this.renderArrayHandler()}</div>;
	}
}

export default Portion1;

Portion1.propTypes = {
	achieversData: shape({
		firstPart: shape({
			heading: string.isRequired,
			text: string.isRequired
		}).isRequired,
		class17Data: shape({
			textYear: string.isRequired,
			data: arrayOf(
				shape({
					id: number,
					textHeading: string.isRequired,
					text: string.isRequired,
					image: string.isRequired
				})
			).isRequired
		}).isRequired
	}).isRequired
};
