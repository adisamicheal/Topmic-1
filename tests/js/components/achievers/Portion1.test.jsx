import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Portion1 from '../../../../src/js/components/achievers/Portion1';

Enzyme.configure({ adapter: new Adapter() });

describe('Test the Portion1 Component', () => {
	const achieversData = {
		firstPart: {
			heading: '',
			text: ''
		},
		class17Data: {
			textYear: '',
			data: [
				{
					id: 1,
					textHeading: '',
					text: '',
					image: ''
				},
				{
					id: 2,
					textHeading: 'Ahmed',
					text: '',
					image: ''
				},
				{
					id: 3,
					textHeading: 'Waqar',
					text: '',
					image: ''
				},
				{
					id: 4,
					textHeading: 'Ahsan',
					text: '',
					image: ''
				},
				{
					id: 5,
					textHeading: 'Hassan',
					text: '',
					image: ''
				}
			]
		},
		class16Data: {
			textYear: '',
			data: [
				{
					id: 1,
					textHeading: '',
					text: '',
					image: ''
				},
				{
					id: 2,
					textHeading: 'Ahmed',
					text: '',
					image: ''
				},
				{
					id: 3,
					textHeading: 'Waqar',
					text: '',
					image: ''
				}
			]
		}
	};

	const wrapper = shallow(<Portion1 achieversData={achieversData} />);
	it('test the rendring', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('test the data in the array', () => {
		expect(achieversData.class17Data.data[0].id).toBe(1);
	});
});
