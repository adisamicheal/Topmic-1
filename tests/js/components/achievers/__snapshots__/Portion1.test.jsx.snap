// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Portion1 Component test the rendring 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Portion1
    achieversData={
      Object {
        "class16Data": Object {
          "data": Array [
            Object {
              "id": 1,
              "image": "",
              "text": "",
              "textHeading": "",
            },
            Object {
              "id": 2,
              "image": "",
              "text": "",
              "textHeading": "Ahmed",
            },
            Object {
              "id": 3,
              "image": "",
              "text": "",
              "textHeading": "Waqar",
            },
          ],
          "textYear": "",
        },
        "class17Data": Object {
          "data": Array [
            Object {
              "id": 1,
              "image": "",
              "text": "",
              "textHeading": "",
            },
            Object {
              "id": 2,
              "image": "",
              "text": "",
              "textHeading": "Ahmed",
            },
            Object {
              "id": 3,
              "image": "",
              "text": "",
              "textHeading": "Waqar",
            },
            Object {
              "id": 4,
              "image": "",
              "text": "",
              "textHeading": "Ahsan",
            },
            Object {
              "id": 5,
              "image": "",
              "text": "",
              "textHeading": "Hassan",
            },
          ],
          "textYear": "",
        },
        "firstPart": Object {
          "heading": "",
          "text": "",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h2>
          
        </h2>
        <div
          className="box"
        />
        <Part
          image=""
          text=""
          textHeading=""
        />
        <Part
          image=""
          text=""
          textHeading="Ahmed"
        />
        <Part
          image=""
          text=""
          textHeading="Waqar"
        />
        <Part
          image=""
          text=""
          textHeading="Ahsan"
        />
        <Part
          image=""
          text=""
          textHeading="Hassan"
        />
        <div
          className="clear"
        />
      </div>,
      "className": "portion1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            
          </h2>,
          <div
            className="box"
          />,
          Array [
            <Part
              image=""
              text=""
              textHeading=""
            />,
            <Part
              image=""
              text=""
              textHeading="Ahmed"
            />,
            <Part
              image=""
              text=""
              textHeading="Waqar"
            />,
            <Part
              image=""
              text=""
              textHeading="Ahsan"
            />,
            <Part
              image=""
              text=""
              textHeading="Hassan"
            />,
          ],
          <div
            className="clear"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "box",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "image": "",
            "text": "",
            "textHeading": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "image": "",
            "text": "",
            "textHeading": "Ahmed",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "image": "",
            "text": "",
            "textHeading": "Waqar",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "image": "",
            "text": "",
            "textHeading": "Ahsan",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "function",
          "props": Object {
            "image": "",
            "text": "",
            "textHeading": "Hassan",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "clear",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h2>
            
          </h2>
          <div
            className="box"
          />
          <Part
            image=""
            text=""
            textHeading=""
          />
          <Part
            image=""
            text=""
            textHeading="Ahmed"
          />
          <Part
            image=""
            text=""
            textHeading="Waqar"
          />
          <Part
            image=""
            text=""
            textHeading="Ahsan"
          />
          <Part
            image=""
            text=""
            textHeading="Hassan"
          />
          <div
            className="clear"
          />
        </div>,
        "className": "portion1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              
            </h2>,
            <div
              className="box"
            />,
            Array [
              <Part
                image=""
                text=""
                textHeading=""
              />,
              <Part
                image=""
                text=""
                textHeading="Ahmed"
              />,
              <Part
                image=""
                text=""
                textHeading="Waqar"
              />,
              <Part
                image=""
                text=""
                textHeading="Ahsan"
              />,
              <Part
                image=""
                text=""
                textHeading="Hassan"
              />,
            ],
            <div
              className="clear"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "box",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "image": "",
              "text": "",
              "textHeading": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "image": "",
              "text": "",
              "textHeading": "Ahmed",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "function",
            "props": Object {
              "image": "",
              "text": "",
              "textHeading": "Waqar",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "function",
            "props": Object {
              "image": "",
              "text": "",
              "textHeading": "Ahsan",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "function",
            "props": Object {
              "image": "",
              "text": "",
              "textHeading": "Hassan",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "clear",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
